packages/@mantine/modals/src/context.ts

```diff
@@ -1,16 +1,20 @@
 import { createContext, ReactNode } from 'react';
-import { ModalProps } from '@mantine/core';
+import { DataAttributes, ModalProps } from '@mantine/core';
 import type { ConfirmModalProps } from './ConfirmModal';
 
 export type ModalSettings = Partial<Omit<ModalProps, 'opened'>> & { modalId?: string };
 
 export type ConfirmLabels = Record<'confirm' | 'cancel', ReactNode>;
 
 export interface OpenConfirmModal extends ModalSettings, ConfirmModalProps {}
-export interface OpenContextModal<CustomProps extends Record<string, any> = {}>
-  extends ModalSettings {
-  innerProps: CustomProps;
-}
+
+export type ContextModalInnerProps<
+  TKey extends MantineModal,
+  P = Parameters<MantineModals[TKey]>[0]['innerProps'],
+> = keyof NonNullable<P> extends never ? { innerProps?: never } : { innerProps: P };
```

Comment:
never can be switched to `any` or `P` depending on the behavior you want.

With `innerProps?: never`: 
openContextModal({ modal: 'myCustomModal', innerProps: {} }); // Type '{}' is not assignable to type 'undefined'.

With `innerProps?: P`: 
openContextModal({ modal: 'myCustomModal', innerProps: {} }); // No error.